### PR TITLE
Add async_io for Java API

### LINKED ISSUE #12183: Java API for async_io
It's the feature of ReadOptions, now available in C api, and java api needs this feature too.

### PR DESCRIPTION
**Pull request by InfiniteNightmare**
Fixes #12183

### CODE REVIEW

**Review by ajkr — APPROVED** [review]

File: java/src/main/java/org/rocksdb/ReadOptions.java [hunk 1]
```diff
@@ -750,6 +750,37 @@ public ReadOptions setValueSizeSoftLimit(final long valueSizeSoftLimit) {
...
+   * Experimental
+   * <p>
+   * If async_io is enabled, RocksDB will prefetch some of data asynchronously.
+   * RocksDB apply it if reads are sequential and its internal automatic
+   * prefetching.
+   * <p>
+   * Default: false
+   * @return true if async_io is enabled.
+   */
+  public boolean asyncIO() {
```
Comment (adamretter) [inline java/src/main/java/org/rocksdb/ReadOptions.java:763]: @InfiniteNightmare If this is experimental it should have the `@Experimental` annotation. Could you add it please?

Comment (adamretter) [reply java/src/main/java/org/rocksdb/ReadOptions.java:763]: Also Java convention dictates that this should be named `asyncIo` and not `asyncIO` please

File: java/src/main/java/org/rocksdb/ReadOptions.java [hunk 2]
```diff
@@ -750,6 +750,37 @@ public ReadOptions setValueSizeSoftLimit(final long valueSizeSoftLimit) {
...
+   * Experimental
+   * <p>
+   * If async_io is enabled, RocksDB will prefetch some of data asynchronously.
+   * RocksDB apply it if reads are sequential and its internal automatic
+   * prefetching.
+   * <p>
+   * @param asyncIO async_io enabled or not.
+   * @return the reference to the current ReadOptions.
+   */
+  public ReadOptions setAsyncIO(final boolean asyncIO) {
```
Comment (adamretter) [inline java/src/main/java/org/rocksdb/ReadOptions.java:778]: Same comments as above please.

File: java/src/main/java/org/rocksdb/ReadOptions.java [hunk 3]
```diff
@@ -817,4 +848,6 @@ private native void setIterateLowerBound(final long handle,
   private native void setIoTimeout(final long handle, final long ioTimeout);
   private native long valueSizeSoftLimit(final long handle);
   private native void setValueSizeSoftLimit(final long handle, final long softLimit);
+  private native boolean asyncIO(final long handle);
```
Comment (adamretter) [inline java/src/main/java/org/rocksdb/ReadOptions.java:851]: Same comments as above please

File: java/src/test/java/org/rocksdb/ReadOptionsTest.java [hunk 4]
```diff
@@ -255,6 +255,14 @@ public void valueSizeSoftLimit() {
     }
   }
 
+  @Test
+  public void asyncIO() {
```
Comment (adamretter) [inline java/src/test/java/org/rocksdb/ReadOptionsTest.java:259]: Same naming comments as before please.

**Review by adamretter — COMMENTED** [review]
Some small changes needed so that it fits with the rest of the RocksJava API.

**Pull request closed without merging**